### PR TITLE
Ordered consumer improvements

### PR DESCRIPTION
**Pull request by scottf**
improved ordered consumer based on work done to implement in .NET

### CODE REVIEW

File: src/main/java/io/nats/client/impl/NatsSubscription.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ class NatsSubscription extends NatsConsumer implements Subscription {
     private Function<NatsMessage, NatsMessage> beforeQueueProcessor;
 
     NatsSubscription(String sid, String subject, String queueName, NatsConnection connection,
-            NatsDispatcher dispatcher) {
+                     NatsDispatcher dispatcher) {
```
Comment (piotrpio) [inline src/main/java/io/nats/client/impl/NatsSubscription.java:39]: Is this formatted correctly rn?

Comment (piotrpio) [inline src/main/java/io/nats/client/impl/NatsSubscription.java:39]: Is this formatted correctly rn?

File: README.md [hunk 2]
```diff
@@ -567,7 +567,7 @@ Subscription creation has many checks to make sure that a valid, operable subscr
 | JsSoOrderedNotAllowedWithDeliverGroup | SO | 90105 | Deliver group is not allowed with an ordered consumer. |
 | JsSoOrderedNotAllowedWithDurable | SO | 90106 | Durable is not allowed with an ordered consumer. |
 | JsSoOrderedNotAllowedWithDeliverSubject | SO | 90107 | Deliver subject is not allowed with an ordered consumer. |
-| JsSoOrderedRequiresAckPolicyNone | SO | 90108 | Deliver subject is not allowed with an ordered consumer. |
+| JsSoOrderedRequiresAckPolicyNone | SO | 90108 | Ordered Requires Ack Policy None. |
```
Comment (piotrpio) [inline README.md:570]:
```suggestion
| JsSoOrderedRequiresAckPolicyNone | SO | 90108 | Ordered Consumer Requires Ack Policy None. |
```

File: src/main/java/io/nats/client/support/NatsJetStreamClientError.java [hunk 3]
```diff
@@ -33,7 +33,7 @@ public class NatsJetStreamClientError {
     public static final NatsJetStreamClientError JsSoOrderedNotAllowedWithDeliverGroup = new NatsJetStreamClientError(SO, 90105, "Deliver group is not allowed with an ordered consumer.");
     public static final NatsJetStreamClientError JsSoOrderedNotAllowedWithDurable = new NatsJetStreamClientError(SO, 90106, "Durable is not allowed with an ordered consumer.");
     public static final NatsJetStreamClientError JsSoOrderedNotAllowedWithDeliverSubject = new NatsJetStreamClientError(SO, 90107, "Deliver subject is not allowed with an ordered consumer.");
-    public static final NatsJetStreamClientError JsSoOrderedRequiresAckPolicyNone = new NatsJetStreamClientError(SO, 90108, "Deliver subject is not allowed with an ordered consumer.");
+    public static final NatsJetStreamClientError JsSoOrderedRequiresAckPolicyNone = new NatsJetStreamClientError(SO, 90108, "Ordered Requires Ack Policy None.");
```
Comment (piotrpio) [inline src/main/java/io/nats/client/support/NatsJetStreamClientError.java:36]:
```suggestion
    public static final NatsJetStreamClientError JsSoOrderedRequiresAckPolicyNone = new NatsJetStreamClientError(SO, 90108, "Ordered Consumer Requires Ack Policy None.");
```

File: README.md (hunk 2 above)
Comment (scottf) [reply README.md:570]: fixed

File: src/main/java/io/nats/client/impl/NatsSubscription.java (hunk 1 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsSubscription.java:39]: fixed

File: src/main/java/io/nats/client/support/NatsJetStreamClientError.java (hunk 3 above)
Comment (scottf) [reply src/main/java/io/nats/client/support/NatsJetStreamClientError.java:36]: fixed

**Review by piotrpio — APPROVED** [review]
LGTM

**Pull request merged**